core/arch/arm/tee/entry_std.c

```diff
@@ -108,14 +147,12 @@ static TEE_Result copy_in_params(const struct optee_msg_param *params,
 
 	for (n = 0; n < num_params; n++) {
 		uint32_t attr;
+		saved_attr[n] = params[n].attr;
 
-		if (params[n].attr & OPTEE_MSG_ATTR_META)
+		if (saved_attr[n] & OPTEE_MSG_ATTR_META)
 			return TEE_ERROR_BAD_PARAMETERS;
-		if (params[n].attr & OPTEE_MSG_ATTR_NONCONTIG)
-			return TEE_ERROR_BAD_PARAMETERS;
```

Comment:
You mean we should disable noncontig bufs as parameters to TA unless it's via already registered buffers? Why?
I think it could be useful for one-shot invokes etc.